### PR TITLE
Use Azure Key Vault to store Postgres password

### PR DESCRIPTION
**Pull request by leorossi**
Refs: #1004

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,5 +32,8 @@
     "docsify-cli": "^4.4.3",
     "husky": "^7.0.1",
     "lerna": "^3.22.1"
+  },
+  "dependencies": {
+    "fastify-secrets-azure": "^1.0.0"
```
Comment (simoneb) [inline package.json:37]: wrong place where to install this dependency

File: packages/titus-backend/.env.sample [hunk 2]
```diff
@@ -45,6 +45,10 @@ SECRETS_PG_PASS=PG_PASS
 #
 # SECRETS_STRATEGY=aws
 # SECRETS_PG_PASS=DB_PASSWORD
-
+#
+# This is an example for Azure Key Vault
+#
+# SECRETS_STRATEGY=azr
```
Comment (simoneb) [inline packages/titus-backend/.env.sample:51]: let's name the strategy azure for consistency

**Review by simoneb — COMMENTED** [review]
Ok the approach looks good, please fix the comments, remove logging, fix linting issues reported by github

File: packages/titus-backend/.env.sample (hunk 2 above)
Comment (leorossi) [reply packages/titus-backend/.env.sample:51]: Will do! (btw I used 3 chars for consistency 😅)

Comment (simoneb) [reply packages/titus-backend/.env.sample]: fair enough, a different definition of consistency. let's stick to the "official" names: gcp, aws, azure

File: package.json [hunk 3]
```diff
@@ -32,5 +32,6 @@
     "docsify-cli": "^4.4.3",
     "husky": "^7.0.1",
     "lerna": "^3.22.1"
-  }
+  },
+  "dependencies": {}
```
Comment (simoneb) [inline package.json:36]: remove!

File: packages/titus-db-manager/lib/secrets-manager.js [hunk 4]
```diff
@@ -26,6 +27,15 @@ async function secretsManager(server) {
...
     )
   }
+
+  if (config.secretsManager.strategy === 'azure') {
+    // add required options
+    const [vaultName, secretName] =
+      config.secretsManager.secrets.dbPassword.split('|')
+    config.secretsManager.secrets.dbPassword = secretName
+    config.secretsManager.clientOptions = { vaultName }
+    console.log(config.secretsManager)
```
Comment (simoneb) [inline packages/titus-db-manager/lib/secrets-manager.js:37]: remove

**Review by simoneb — CHANGES_REQUESTED** [review]
see comments

**Review by simoneb — APPROVED** [review]
LGTM. Next step would be to change the Azure deployment configuration to use this and make sure it works.

**Pull request merged**